### PR TITLE
Migrate BTV pacakges to new PoolDBOutputService

### PR DESCRIPTION
**Pull request by panwarlsweet**
#### PR description:
As introduced in issue https://github.com/cms-AlCaDB/AlCaTools/issues/28, migrating to the new methods replacing writeOne --> writeOneIOV, createNewIOV --> createOneIOV, and appendSinceTime --> appendOneIOV.
The files containing PoolDBOutputService in BTV packages that were addressed are:
RecoBTag/PerformanceDB/plugins/PhysicsPerformanceDBWriterFromFile_WPandPayload.cc
RecoBTag/PerformanceDB/plugins/PhysicsPerformanceDBWriterFromFile_WPandPayload_IOV.cc
RecoBTag/PerformanceDB/plugins/PhysicsPerformanceDBWriterTFormula_fromfile_WPandPL.cc
RecoBTag/TrackProbability/test/CalibrationSkeleton.cc
RecoBTag/TrackProbability/test/TrackProbabilityXMLtoDB.cc

#### PR validation:
- changes compile with CMSSW
- `scram b runtests` works fine

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
N/A

@soureek @johnalison

### CODE REVIEW

**Comment by tvami** [comment]
assign db

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc [hunk 1]
```diff
@@ -81,93 +77,83 @@ using namespace reco;
...
+      calibration.data.push_back(entry);
+    }
+    for (unsigned int i = 0; i < data2d.size(); i++) {
+      TrackProbabilityCalibration::Entry entry;
+      entry.category = data2d[i].first.categoryData();
+      entry.histogram = data2d[i].second;
+      calibration2d.data.push_back(entry);
+    }
+
+    mydbservice->createOneIOV(calibration, "BTagTrackProbability3DRcd");
```
Comment (tvami) [inline RecoBTag/TrackProbability/test/CalibrationSkeleton.cc:135]:
```suggestion
    mydbservice->createOneIOV(calibration, mydbservice->endOfTime(), "BTagTrackProbability3DRcd");
```

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc [hunk 2]
```diff
@@ -81,93 +77,83 @@ using namespace reco;
...
+
+    mydbservice->createOneIOV(calibration2d, "BTagTrackProbability2DRcd");
```
Comment (tvami) [inline RecoBTag/TrackProbability/test/CalibrationSkeleton.cc:137]:
```suggestion
    mydbservice->createOneIOV(calibration2d, mydbservice->endOfTime(), "BTagTrackProbability2DRcd");
```

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc [hunk 3]
```diff
@@ -81,93 +77,83 @@ using namespace reco;
...
-            if(newBinning) data[i].second = CalibratedHistogram(1000, 0, 50);
-            else           data[i].second = CalibratedHistogram();
-          }
-	for(unsigned int i = 0 ; i < data2d.size();i++)
-          {
-            if(newBinning) data2d[i].second = CalibratedHistogram(1000, 0, 50);
-            else           data2d[i].second = CalibratedHistogram();
-          }
-	
+    calibrationNew =
```
Comment (tvami) [inline RecoBTag/TrackProbability/test/CalibrationSkeleton.cc:88]: I dont see where `calibrationNew` is deleted. But this file is under \test so fine...

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc [hunk 4]
```diff
@@ -179,109 +165,83 @@ class CalibrationSkeleton : public edm::EDAnalyzer {
...
-  
+
   //***********************************************************************************
   //mandatory for ip significance reco
-  const TransientTrackBuilder * m_transientTrackBuilder_producer;
+  const TransientTrackBuilder* m_transientTrackBuilder_producer;
   edm::ESHandle<TransientTrackBuilder> builder;
-  iSetup.get<TransientTrackRecord>().get("TransientTrackBuilder",builder);
+  iSetup.get<TransientTrackRecord>().get("TransientTrackBuilder", builder);
   m_transientTrackBuilder_producer = builder.product();
```
Comment (tvami) [inline RecoBTag/TrackProbability/test/CalibrationSkeleton.cc:207]:
```suggestion
auto const m_transientTrackBuilder_producer = &iSetup.getData(transientTrackBuilderToken_)
```
and you'll need to define `transientTrackBuilderToken_` in the header as
``` 
const edm::ESGetToken<TransientTrackBuilder, TransientTrackRecord> transientTrackBuilderToken_;
```

File: RecoBTag/TrackProbability/test/TrackProbabilityXMLtoDB.cc [hunk 5]
```diff
@@ -78,82 +74,70 @@ using namespace reco;
 //
 
 class TrackProbabilityXMLtoDB : public edm::EDAnalyzer {
```
Comment (tvami) [inline RecoBTag/TrackProbability/test/TrackProbabilityXMLtoDB.cc:76]:
```suggestion
class TrackProbabilityXMLtoDB : public edm::one::EDAnalyzer<> {
```

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc [hunk 6]
```diff
@@ -81,93 +77,83 @@ using namespace reco;
 //
 
 class CalibrationSkeleton : public edm::EDAnalyzer {
```
Comment (tvami) [inline RecoBTag/TrackProbability/test/CalibrationSkeleton.cc:79]:
```suggestion
class CalibrationSkeleton : public edm::one::EDAnalyzer<> {
```

**Comment by tvami** [comment]
Hi @panwarlsweet thanks for doing this. The AlCaDB migration looks good, but then there is the es consume migration and the legacy module migration that needs to be taken care of. I gave suggestions above.

**Comment by tvami** [comment]
@cms-sw/reconstruction-l2 should these codes be moved under CondTools? I'm not sure if it's best under reco, what do you think?

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc (hunk 1 above)
Comment (panwarlsweet) [reply RecoBTag/TrackProbability/test/CalibrationSkeleton.cc]: corrected

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc (hunk 2 above)
Comment (panwarlsweet) [reply RecoBTag/TrackProbability/test/CalibrationSkeleton.cc]: corrected

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc (hunk 4 above)
Comment (panwarlsweet) [reply RecoBTag/TrackProbability/test/CalibrationSkeleton.cc]: implemented as suggested above

File: RecoBTag/TrackProbability/test/TrackProbabilityXMLtoDB.cc (hunk 5 above)
Comment (panwarlsweet) [reply RecoBTag/TrackProbability/test/TrackProbabilityXMLtoDB.cc]: corrected

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc (hunk 6 above)
Comment (panwarlsweet) [reply RecoBTag/TrackProbability/test/CalibrationSkeleton.cc]: corrected

**Comment by slava77** [comment]
> @cms-sw/reconstruction-l2 should these codes be moved under CondTools? I'm not sure if it's best under reco, what do you think?

fine with me

**Comment by tvami** [comment]
> > @cms-sw/reconstruction-l2 should these codes be moved under CondTools? I'm not sure if it's best under reco, what do you think?
> 
> fine with me

Hi @panwarlsweet should this move be done in this PR, or you prefer to do that in a different one. 
Also can you please squash the commits into one? Thanks!

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by soureek** [comment]
> @cms-sw/reconstruction-l2 should these codes be moved under CondTools? I'm not sure if it's best under reco, what do you think?
@tvami These codes are traditionally maintained in the RecoBTag package and the BTV group makes update time to time. Is it really a strong requirement to move them under CondTools? One also has to consider that other codes used by BTV is then required to be modified if the classes are moved elsewhere. This might not be trivial to do in a short time.

**Comment by tvami** [comment]
> @tvami These codes are traditionally maintained in the RecoBTag package and the BTV group makes update time to time.

It would be still BTV responsibility of course. The point is that other groups have there own packages under CondTools for code that interacts with the DB. So you could have your own CondTools/BTag there.

 > Is it really a strong requirement to move them under CondTools?

No, it's not. And I guess that's the reason for it not to happen so far haha

> One also has to consider that other codes used by BTV are then required to be modified if the classes are moved elsewhere. This might not be trivial to do in a short time.

Sure, an LXR search should make this easy tho. Anyway, we can sign the PR for the time being

**Comment by tvami** [comment]
+db
 * agree to move the files in a new PR
 * tests pass
 * code in line with describtion

**Comment by slava77** [comment]
@panwarlsweet @tvami 
I just wanted to check if the relocation of the files is expected in this PR or deferred to the next.
Please clarify.
Thank you.

File: TrackingTools/TransientTrack/interface/TransientTrackBuilder.h [hunk 7]
```diff
@@ -73,6 +74,7 @@ class TransientTrackBuilder {
 private:
   const MagneticField* theField;
   edm::ESHandle<GlobalTrackingGeometry> theTrackingGeometry;
+  edm::ESGetToken<TransientTrackBuilder, TransientTrackRecord> transientTrackBuilderToken_;
```
Comment (slava77) [inline TrackingTools/TransientTrack/interface/TransientTrackBuilder.h:77]: I'm not sure how the change in this file belongs in this PR.

File: RecoBTag/TrackProbability/test/CalibrationSkeleton.cc [hunk 8]
```diff
@@ -179,109 +165,80 @@ class CalibrationSkeleton : public edm::EDAnalyzer {
...
-  
-  
+
   //***********************************************************************************
   //mandatory for ip significance reco
-  const TransientTrackBuilder * m_transientTrackBuilder_producer;
-  edm::ESHandle<TransientTrackBuilder> builder;
-  iSetup.get<TransientTrackRecord>().get("TransientTrackBuilder",builder);
-  m_transientTrackBuilder_producer = builder.product();
+  auto const m_transientTrackBuilder_producer = &iSetup.getData(transientTrackBuilderToken_);
```
Comment (slava77) [inline RecoBTag/TrackProbability/test/CalibrationSkeleton.cc:204]: does this code even compile?
I'm not sure where `transientTrackBuilderToken_` is coming from.

This file is apparently not compiled, based on RecoBTag/TrackProbability/test/BuildFile.xml

Comment (slava77) [reply RecoBTag/TrackProbability/test/CalibrationSkeleton.cc:204]: even though requirements on the code in the /test directory are fairly light, I'd still rather that the code at least compiles.

The modification in TrackingTools/TransientTrack/interface/TransientTrackBuilder.h should be reverted (it doesn't make much sense in that context).

**Comment by slava77** [comment]
@panwarlsweet 
please clarify on the status/plans of updating this PR following https://github.com/cms-sw/cmssw/pull/36231#pullrequestreview-821876106
Thank you.

**Comment by panwarlsweet** [comment]
Hello @slava77 

I am sorry for the delayed reply. It seems the particular code in file CalibrationSkeleton.cc is quite outdated and can be dropped altogether. Therefore, I will revert back the changes I made to CalibrationSkeleton.cc and TransientTrackBuilder.h so that PR could be merged.

**Comment by tvami** [comment]
Hi @panwarlsweet if you did these change, I dont think you've pushed them here, or at least I dont see the commit.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+db
 * there is more to do with these packages (move them to a correct directory, migrate away from es get()) but it will be done in a follow-up PR
 * Jenkins tests pass

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/36231	ad6e3e2 
- code changes are technical, specific to the PoolDB interface update 
- jenkins tests pass
- I expect that the warnings related to es.get will be covered in a separate PR

**Comment by qliphy** [comment]
+1

**Pull request merged**